### PR TITLE
Add translator for 知乎Zhihu

### PR DESCRIPTION
**Pull request by l0o0**
[知乎Zhihu](https://www.zhihu.com) is a Quora-like site in China. This translator can save answers from the question page and search results. The item type is webpage, and the detailed post is saved to a note

### CODE REVIEW

**Comment by l0o0** [comment]
@AbeJellinek So sorry for the late reply. I've updated the code.

File: Zhihu.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	let m = url.match(/\/(answer|p)\/(\d+)/);
+	if (!m) return false;
+	return { ztype: m[1] === 'p' ? 'article' : 'answer', zid: m[2], url: url };
+}
+
+function detectWeb(doc, url) {
+	let ZID = getIDFromUrl(url);
+	// Z.debug(ZID);
+	if (ZID) {
+		return "webpage";
```
Comment (AbeJellinek) [inline Zhihu.js:55]: Would `forumPost` be more appropriate?

**Review by AbeJellinek — CHANGES_REQUESTED** [review]

File: Zhihu.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	}
+	return false;
+}
+
+function getSearchResults(url, doc, checkOnly, itemInfo) {
+	if (checkOnly
+		&& (url.includes("search?type=")
+			|| url.includes("/collection/")
+			|| url.includes("/people/"))
+	) return true;
```
[the remark this answers is not in the document]
Comment (l0o0) [reply Zhihu.js:68]: Yes, your are right. The code below can return multiple results. These codes are added deliberately. Sometimes translator can not return multiple results in search, like https://www.zhihu.com/search?type=content&q=zotero). So I use url to return multiple.

File: Zhihu.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+		else { // 问题页,首页,专栏
+			data = JSON.parse(data);
+			url = rows[i].querySelectorAll("meta[itemprop='url']");
+			url = url[url.length - 1].getAttribute('content').replace(/^\/\//, 'https://');
+			ZID = { ztype: data.type, zid: data.itemId, url: url };
+			title = i + ' ' + data.authorName + ' : ' + data.title;
+		}
+		if (checkOnly) return true;
+		found = true;
+		itemInfo[url] = ZID;
```
[the remark this answers is not in the document]
Comment (l0o0) [reply Zhihu.js:90]: I have updated these codes, new codes will be pushed tomorrow

**Comment by l0o0** [comment]
@AbeJellinek Hi, codes are updated. Thank you for your detailed review.

**Pull request closed without merging**